src/python/TaskWorker/Actions/DagmanCreator.py

```diff
@@ -366,12 +366,19 @@ def __init__(self, config, crabserver, procnum=-1, rucioClient=None):
     def populateGlideinMatching(self, info):
         scram_arch = info['tm_job_arch']
         # Set defaults
-        info['desired_arch'] = "X86_64"
+        info['required_arch'] = "X86_64"
+        # The following regex matches a scram arch into four groups 
+        # for example el9_amd64_gcc10 is matched as (el)(9)_(amd64)_(gcc10)
+        # later, only the third group is returned, the one corresponding to the arch.
         m = re.match("([a-z]+)(\d+)_(\w+)_(\w+)", scram_arch)
         if m:
             _, _, arch, _ = m.groups()
-            if arch == "amd64":
-                info['desired_arch'] = "X86_64"
+            if arch not in SCRAM_TO_ARCH:
+                msg = f"Job configured to a ScramArch: '{arch}' not supported in TaskWorker"
```

Comment:
let me be picky, the `as` should not be there. I suggest `to a ScramArch : not...` -->` for non-supporterd ScramArch`